pkg/gitparse/gitparse.go

```diff
@@ -622,15 +681,51 @@ func isMessageEndLine(isStaged bool, latestState ParseState, line []byte) bool {
 	return false
 }
 
+// `Notes:` or `Notes (context):`
+// See https://tylercipriani.com/blog/2022/11/19/git-notes-gits-coolest-most-unloved-feature/
+func isNotesStartLine(isStaged bool, latestState ParseState, line []byte) bool {
+	if isStaged || latestState != MessageEndLine {
+		return false
+	}
+	if len(line) > 5 && bytes.Equal(line[:5], []byte("Notes")) {
+		return true
+	}
+	return false
+}
+
+// Line after NotesStartLine that starts with 4 spaces
+func isNotesLine(isStaged bool, latestState ParseState, line []byte) bool {
+	if isStaged || !(latestState == NotesStartLine || latestState == NotesLine) {
+		return false
+	}
+	if len(line) > 4 && bytes.Equal(line[:4], []byte("    ")) {
+		return true
+	}
+	return false
+}
+
+// Line directly after NotesLine with only a newline.
+func isNotesEndLine(isStaged bool, latestState ParseState, line []byte) bool {
+	if isStaged || latestState != NotesLine {
+		return false
+	}
+	if len(strings.TrimRight(string(line[:]), "\r\n")) == 0 {
+		return true
+	}
+	return false
+}
+
 // diff --git a/internal/addrs/move_endpoint_module.go b/internal/addrs/move_endpoint_module.go
 func isDiffLine(isStaged bool, latestState ParseState, line []byte) bool {
 	if !(latestState == MessageStartLine || // Empty commit messages can go from MessageStart->Diff
 		latestState == MessageEndLine ||
+		latestState == NotesEndLine ||
 		latestState == BinaryFileLine ||
+		latestState == ModeLine ||
 		latestState == IndexLine ||
 		latestState == HunkContentLine ||
 		latestState == ParseFailure) {
-		if latestState == Initial && !isStaged {
+		if !(isStaged && latestState == Initial) {
```

Comment:
This is an unrelated bug that caused `isDiffLine` to return true for any `latestState`, not just the ones mentioned in the list.

Reply:
Was there any real impact of that? Or it would it only flare up if the state machine got into a bad state in some other way?